src/perf/best-practices.md

```diff
@@ -0,0 +1,411 @@
+[`ColorFilter`]: {{site.api}}/flutter/dart-ui/ColorFilter-class.html
+[`FadeInImage`]: {{site.api}}/flutter/widgets/FadeInImage-class.html
+[`Opacity`]: {{site.api}}/flutter/widgets/Opacity-class.html
+[`ShaderMask`]: {{site.api}}/flutter/widgets/ShaderMask-class.html
+[`Text`]: {{site.api}}/flutter/widgets/Text-class.html
+[Transparent image]: {{site.api}}/flutter/widgets/Opacity-class.html#transparent-image
+
+---
+
+## Minimize use of opacity and clipping
+
+Opacity is another expensive operation, as is clipping.
+Here are some tips you might find to be useful:
+
+* Use the [`Opacity`][] widget only when necessary.
+  See the [Transparent image][] section in the `Opacity`
+  API page for an example of applying opacity directly
+  to an image, which is faster than using the `Opacity`
+  widget.
+* Instead of wrapping simple shapes or text
+  in an `Opacity` widget, it's usually faster to
+  just draw them with a semitransparent color.
+  (Though this only works if there are no overlapping
+  bits in the to-be-drawn shape.)
+* To implement fading in an image, consider using the
+  [`FadeInImage`][] widget, which applies a gradual
+  opacity using the GPU’s fragment shader.
+  For more information, check out the [`Opacity`][] docs.
+* **Clipping** doesn’t call `saveLayer()` (unless
+  explicitly requested with `Clip.antiAliasWithSaveLayer`),
+  so these operations aren’t as expensive as `Opacity`,
+  but clipping is still costly, so use with caution.
+  By default, clipping is disabled (`Clip.none`),
+  so you must explicitly enable it when needed.
+* To create a rectangle with rounded corners,
+  instead of applying a clipping rectangle,
+  consider using the `borderRadius` property offered
+  by many of the widget classes.
+
+---
+
+## Implement grids and lists thoughtfully 
+
+How your grids and lists are implemented
+might be causing performance problems for your app.
+This section describes an important best
+practice when creating grids and lists,
+and how to determine whether your app uses
+excessive layout passes.
+
+### Be lazy!
+
+When building a large grid or list,
+use the lazy builder methods, with callbacks.
+That ensures that only the visible portion of the
+screen is built at startup time.
+
+For more information and examples, check out:
+
+* [Working with long lists][] in the [Cookbook][]
+* [Creating a `ListView` that loads one page at a time][]
+  a community article by AbdulRahman AlHamali
+* [`Listview.builder`][] API
+
+[Cookbook]: {{site.url}}/cookbook
+[Creating a `ListView` that loads one page at a time]: {{site.medium}}/saugo360/flutter-creating-a-listview-that-loads-one-page-at-a-time-c5c91b6fabd3
+[`Listview.builder`]: {{site.api}}/flutter/widgets/ListView/ListView.builder.html
+[Working with long lists]: {{site.url}}/cookbook/lists/long-lists
+
+<a href="minimize-layout-passes-caused-by-intrinsic-operations"></a>
+### Minimize layout passes caused by intrinsic operations
```

Comment:
should this be ## instead of ###? I don't think minimizing layout passes caused by intrinsic operations applies explicitly to Lists and Grids. @goderbauer can you check me on this?